recipes/libtommath/all/conanfile.py

```diff
@@ -21,119 +28,72 @@ class LibTomMathConan(ConanFile):
         "fPIC": True,
     }
 
-    exports_sources = "patches/*"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
-    def build_requirements(self):
-        if self._settings_build.os == "Windows" and self.settings.compiler != "Visual Studio":
-            self.build_requires("make/4.3")
-        if self.settings.compiler != "Visual Studio" and self.settings.os != "Windows" and self.options.shared:
-            self.build_requires("libtool/2.4.6")
+        if is_msvc(self) and self.settings.build_type == "Debug":
+            # libtommath requires at least /O1 on MSVC for dead code elimination
+            # https://github.com/libtom/libtommath/blob/42b3fb07e7d504f61a04c7fca12e996d76a25251/s_mp_rand_platform.c#L120-L138
+            self.output.warning("Debug builds on MSVC are not supported, switching to RelWithDebInfo")
+            self.settings.build_type = "RelWithDebInfo"
```

Comment:
Hi! Thanks a lot for fixing the failing configuration. I'll talk with the team about your proposed solution, which while interesting, it's better if it has all our consensus on going ahead with it :)